content/articles/client-server-react-django-recaptcha/index.md

```diff
@@ -0,0 +1,357 @@
+  const [email, setEmail] = useState('')
+  const [message, setMessage] = useState('')
+  const [captchaResult, setCaptchaResult] = useState()
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+
+    // Handle form submission here
+
+  }
+
+     const handleRecaptcha = (value) => {
+
+       fetch('http://127.0.0.1:8000/recaptcha/', {
+         method: 'POST',
+         body: JSON.stringify({ 'captcha_value': value }),
+         headers: { 'Content-Type': 'application/json' }
+       })
+        .then(res => res.json())
+        .then(data => {
+          console.log(data.captcha.success)
+          setCaptchaResult(data.captcha.success)
+        }) 
+     }
+  
+
+
+  return (
+    <div className="container">
+      <h2>Contact us:</h2>
+      <form>
+        <label>Name</label>
+        <input type="text" name="name" placeholder="Your name.." value={name} onChange={(e) => {setName(e.target.value)}} />
+
+        <label>Email</label>
+        <input type="email" name="email" placeholder="example@domain.com" value={email} onChange={(e) => {setEmail(e.target.value)}} />
+
+        <label>Subject</label>
+        <textarea name="subject" placeholder="Write something.." value={message} onChange={(e) => {setMessage(e.target.value)}} />
+
+        <div className="cta">
+          <ReCAPTCHA
+            sitekey="YOUR SITE KEY"
+            onChange={handleRecaptcha}
+          />
+          
+          {
+             captchaResult && <button type='submit' onClick={(e) => {handleSubmit(e)}}>Submit</button>
+          }
+        </div>
+  </form>
+</div>
+  );
+}
+
+export default App;
+```
+
+**Code explanation**
+
+- In the above code, we have a contact form that contains the fields; name, email, and message. We keep track of the data entered in these fields using the useState() hook.
+- In the `div` with the className `cta`, we have our ReCaptcha component that takes two attributes, one is the site key that we get from the Google reCAPTCHA website. The other attribute is an `onChange` attribute that calls the `handleRecaptcha` function when the reCAPTCHA component is clicked on.
+- Below the reCAPTCHA component, we conditionally render the `Submit` button depending on the outcome of the reCAPTCHA. We only show it when the success value of the reCAPTCHA evaluates to true(when the user successfully solves the reCAPTCHA challenge). This helps to ensure that the form data can only be submitted after the reCAPTCHA challenge has been solved.
+- The `handleRecaptcha` function takes in a value that we send to the backend for server-side verification. You can console.log the value to see it. We use the Fetch API to make a POST request to the route `http://127.0.0.1:8000/recaptcha/` which we will, later on, create on our server. We then use the response we get from the server and pass it to the `setCaptchaResult` function which sets the state of the `captchaResult`. The response is either true or false. If it is true, we show the `Submit` button.
```

Comment:
Add the opening and closing brackets for method definitions, i.e `handleRecaptcha()`.Update where applicable!